### PR TITLE
feat: add `order_by` query parameter to NFT token API endpoints

### PR DESCRIPTION
**Pull request by Benzbeeb**
## Summary by CodeRabbit

- New Features
  - Added order_by query parameter to NFT listing endpoints (by account, by collection) to sort results by token_id or height; pagination and cursors respect the chosen sort and order.

- Documentation
  - API docs and Swagger updated with the new order_by parameter, allowed values (token_id, height), and default (token_id).

- Tests
  - Added extensive tests for order_by validation, sorting behavior, pagination integration, edge cases, and error messages.

### CODE REVIEW

File: api/handler/common/pagination.go [hunk 1]
```diff
@@ -293,43 +293,41 @@ func (p *Pagination) ApplyToNftCollection(query *gorm.DB) *gorm.DB {
...
-func (p *Pagination) ApplyToNft(query *gorm.DB) *gorm.DB {
+func (p *Pagination) ApplyToNft(query *gorm.DB, orderBy string) *gorm.DB {
 	switch p.CursorType {
 	case CursorTypeComposite:
 		height, err := p.safeGetInt64("height")
 		if err != nil {
-			// Fallback to offset-based pagination on error
-			return query.Order(p.OrderBy("height", "token_id")).Offset(p.Offset).Limit(p.Limit)
+			if orderBy == "height" {
+				return query.Order(p.OrderBy("height", "token_id")).Offset(p.Offset).Limit(p.Limit)
```
Comment (songwongtp) [inline api/handler/common/pagination.go:302]: let find a way to combine this return with the default option

**Review by songwongtp — APPROVED** [review]
